From 846e15914973e0121b88144a5c9a70ea00f4ad4c Mon Sep 17 00:00:00 2001
From: Feng Zhang <feng@radxa.com>
Date: Tue, 12 May 2026 15:10:17 +0800
Subject: [PATCH] cmd: sunxi: add ARISC boot command

Signed-off-by: Feng Zhang <feng@radxa.com>

diff --git a/src/cmd/Kconfig b/cmd/Kconfig
index 322ebe600c5..b359272b1f7 100644
--- a/src/cmd/Kconfig
+++ b/src/cmd/Kconfig
@@ -3202,4 +3202,8 @@ config CMD_SPAWN_NUM_JOBS
 	  When a jobs exits, its identifier is available to be re-used by the next
 	  spawn command.
 
+config CMD_SUNXI_BOOT_ARISC
+		bool "Load and run arisc cpu"
+		default MACH_SUN60I_A733
+
 endif
diff --git a/src/cmd/Makefile b/cmd/Makefile
index 4cd13d4fa6e..63ac9572319 100644
--- a/src/cmd/Makefile
+++ b/src/cmd/Makefile
@@ -12,6 +12,7 @@ obj-y += panic.o
 obj-y += version.o
 
 # command
+obj-$(CONFIG_CMD_SUNXI_BOOT_ARISC) += sunxi_boot_arisc.o
 obj-$(CONFIG_CMD_ARMFFA) += armffa.o
 obj-$(CONFIG_CMD_2048) += 2048.o
 obj-$(CONFIG_CMD_ACPI) += acpi.o
diff --git a/src/cmd/sunxi_boot_arisc.c b/cmd/sunxi_boot_arisc.c
new file mode 100644
index 00000000000..d4fa0e08160
--- /dev/null
+++ b/src/cmd/sunxi_boot_arisc.c
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-2.0+
+
+#include <command.h>
+#include <cpu_func.h>
+#include <linux/string.h>
+#include <linux/arm-smccc.h>
+#include <linux/types.h>
+
+static u32 sunxi_smc_call_atf(ulong arg0, ulong arg1, ulong arg2, ulong arg3, ulong pResult)
+{
+	struct arm_smccc_res param = { 0 };
+
+	arm_smccc_smc(arg0, arg1, arg2, arg3, 0, 0, 0, 0, &param);
+
+	return param.a0;
+}
+
+static int do_sunxi_boot_arisc(struct cmd_tbl *cmdtp, int flag, int argc,
+		   char *const argv[])
+{
+	u32 ARM_SVC_ARISC_STARTUP = 0x8000ff10;
+
+	sunxi_smc_call_atf(ARM_SVC_ARISC_STARTUP, 0, 0, 0, 0);
+
+	return 0;
+}
+
+U_BOOT_CMD(
+	sunxi_boot_arisc, CONFIG_SYS_MAXARGS, 1, do_sunxi_boot_arisc,
+	"Load and run arisc cpu",
+	"\n"
+);
-- 
2.43.0

